pkg/koordlet/statesinformer/impl/states_nodemetric.go

```diff
@@ -447,12 +447,19 @@ func (r *nodeMetricInformer) collectNodeMetric(queryparam metriccache.QueryParam
 	}
 
 	var memAggregateResult metriccache.AggregateResult
+	nodeMemoryCollectPolicy := *r.getNodeMetricSpec().CollectPolicy.NodeMemoryCollectPolicy
 	// report usageMemoryWithHotPageCache
-	if *r.getNodeMetricSpec().CollectPolicy.NodeMemoryCollectPolicy == slov1alpha1.UsageWithHotPageCache && system.GetIsStartColdMemory() {
+	if nodeMemoryCollectPolicy == slov1alpha1.UsageWithHotPageCache && system.GetIsStartColdMemory() {
```

Comment:
recommend leaving the default case ahead, i.e. `len(policy) <= 0 || policy == UsageWithoutPageCache`